### PR TITLE
Renamed refbox terms to gamecontroller terms

### LINKED ISSUE #1559: Adapt codebase to ssl-game-controller
### Description of the task
ssl-game-controller replaced ssl-refbox since Robocup 2019, so we should adapt to that by renaming our codebase to the new terminology and update the protobufs

https://github.com/RoboCup-SSL/ssl-game-controller

### Acceptance criteria

- [ ] rename refbox to gamecontroller
- [ ] update any terminology to be consistent 
- [ ] Rename refbox gamestate to gamecontroller command

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
* Renamed refbox stage to referee stage
* Renamed RefboxGamestate to RefereeCommand
* Renamed refbox to gamecontroller
* Refactored refbox_data into game_state 
* Ported new gamecontroller proto definitions

### Resolved Issues
resolves #1559

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by kelvinkoon — DISMISSED** [review]
Doing god's work again :pray:

**Review by HaomiaoZ — DISMISSED** [review]
Great Work 🔥

File: src/software/world/world.h [hunk 1]
```diff
@@ -59,27 +58,26 @@ class World final
     void updateEnemyTeamState(const Team& new_enemy_team_data);
 
     /**
-     * Updates the refbox game state
+     * Updates the referee command
      *
-     * @param game_state the game state sent by refbox
+     * @param command the game state sent by the referee
```
Comment (LiCody) [inline src/software/world/world.h]:
```suggestion
     * @param command the command sent by the referee
```

File: src/software/parameter/config/ai_control.yaml [hunk 2]
```diff
@@ -30,23 +30,23 @@ AIControlConfig:
...
-        Specifies the refbox game state that should be in use
-  PreviousRefboxGameState:
+        Specifies the referee command that should be in use
+  PreviousRefereeCommand:
     type: "std::string"
     default: "Halt"
     description: >-
-        Specifies the previous refbox game state to correctly set up 
+        Specifies the previous referee command to correctly set up 
         the game state state machine when overriding game state. For
```
Comment (LiCody) [inline src/software/parameter/config/ai_control.yaml]: This should probably be changed

File: src/software/proto/message_translation/ssl_geometry.h [hunk 3]
```diff
@@ -126,21 +127,22 @@ std::optional<SSL_FieldLineSegment> findLineSegment(
...
-std::optional<SSL_FieldCircularArc> findCircularArc(
-    const google::protobuf::RepeatedPtrField<SSL_FieldCircularArc>& circular_arcs,
+std::optional<SSLProto::SSL_FieldCircularArc> findCircularArc(
+    const google::protobuf::RepeatedPtrField<SSLProto::SSL_FieldCircularArc>&
+        circular_arcs,
     SSLCircularArcs arc_type);
 
 /**
- * Creates a Vector2f proto message from the given point
+ * Creates a SSLProto::Vector2f proto message from the given point
```
Comment (LiCody) [inline src/software/proto/message_translation/ssl_geometry.h:136]: Not sure if we should have the namespace name in the comments

**Review by LiCody — CHANGES_REQUESTED** [review]
LGTM just forgot to change a few comments

File: src/software/world/world.h (hunk 1 above)
Comment (jonathanlew) [reply src/software/world/world.h]: good catch, fixed

File: src/software/parameter/config/ai_control.yaml (hunk 2 above)
Comment (jonathanlew) [reply src/software/parameter/config/ai_control.yaml]: good catch, fixed

File: src/software/proto/message_translation/ssl_geometry.h (hunk 3 above)
Comment (jonathanlew) [reply src/software/proto/message_translation/ssl_geometry.h:136]: hm, I think it increases clarity

**Review by MathewMacDougall — DISMISSED** [review]

**Review by LiCody — DISMISSED** [review]

**Review by kelvinkoon — DISMISSED** [review]

**Review by HaomiaoZ — DISMISSED** [review]

**Review by MathewMacDougall — DISMISSED** [review]

**Review by LiCody — DISMISSED** [review]

**Review by kelvinkoon — DISMISSED** [review]
LGTM pending CI

**Review by MathewMacDougall — APPROVED** [review]

**Review by kelvinkoon — APPROVED** [review]
"aah build errors" :laughing:

**Pull request merged**